### PR TITLE
add option to favourite event

### PR DESCRIPTION
**Pull request by simarsingh24**
initial steps for : #117

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Instead, create another table called favorite events and save IDs of favorite events in it

**Comment by simarsingh24** [comment]
@iamareebjamal is this the right way?

**Comment by simarsingh24** [comment]
@nikit19 still in WIP not near to done

**Comment by iamareebjamal** [comment]
Yes, see this for how to load events which are favorite https://developer.android.com/reference/android/arch/persistence/room/Relation

**Comment by simarsingh24** [comment]
followed the blog, how does it look now @iamareebjamal

File: app/src/main/java/org/fossasia/openevent/general/event/Event.kt [hunk 1]
```diff
@@ -61,5 +61,8 @@ data class Event(
         val isSessionsSpeakersEnabled: Boolean = false,
         val isTicketingEnabled: Boolean = false,
         val isTaxEnabled: Boolean = false,
-        val isMapShown: Boolean = false
+        val isMapShown: Boolean = false,
+
+        @Relation(parentColumn = "id", entityColumn = "eventId")
+        val favoriteEventsList: List<Favourite>
```
[the remark this answers is not in the document]
Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/event/Event.kt]: Right! Favourite table should have list of events

File: app/src/main/java/org/fossasia/openevent/general/event/EventDao.kt [hunk 2]
```diff
@@ -22,4 +22,8 @@ interface EventDao {
 
     @Query("SELECT * from Event WHERE id = :id")
     fun getEvent(id: Long): Flowable<Event>
+
+    @Query("SELECT favoriteEventsList from Event")
+    fun loadFavouriteEvents(): List<Event>
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDao.kt]: this should return all the favourite events @iamareebjamal ?

**Comment by simarsingh24** [comment]
@iamareebjamal  please look at it once and let me know of your views on this

**Comment by simarsingh24** [comment]
@iamareebjamal thatsa how i did it in the first place, but you asked to seperate and create a favourite table
I will add a boolean field to event now, isFavourite

**Comment by iamareebjamal** [comment]
Yes I know, but seeing the complexity you guys are facing with foreign keys and handling database changes, it's better to do things simpler way

Correct way would be to normalize the database, but we need to move things at a faster pace

**Comment by simarsingh24** [comment]
videotogif_2018 06 17_00 05 44
@iamareebjamal screen is flashing on clicking fab..
log ``` isRecyclable decremented below 0: unmatched pair of setIsRecyable() calls for ViewHolder```

**Comment by nikit19** [comment]
@simarsingh24 it could be due to the recyclerView animations try running the app without the animation

**Comment by simarsingh24** [comment]
@nikit19 yes looks like that was the issue 👍

**Comment by iamareebjamal** [comment]
This library is pretty buggy, please remove it in a separate PR

**Comment by simarsingh24** [comment]
@iamareebjamal sure! creating an issue

**Comment by simarsingh24** [comment]
@nikit19 @dreadpool2 @iamareebjamal please review

File: app/src/main/java/org/fossasia/openevent/general/event/Event.kt [hunk 3]
```diff
@@ -61,5 +61,6 @@ data class Event(
         val isSessionsSpeakersEnabled: Boolean = false,
         val isTicketingEnabled: Boolean = false,
         val isTaxEnabled: Boolean = false,
-        val isMapShown: Boolean = false
+        val isMapShown: Boolean = false,
+        val isFavourite: Boolean = false
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/Event.kt]: favorite

File: app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt [hunk 4]
```diff
@@ -8,36 +8,47 @@ import kotlinx.android.synthetic.main.item_card_events.view.*
...
             sendIntent.type = "text/plain"
             itemView.context.startActivity(Intent.createChooser(sendIntent, "Share Event Details"))
         }
+
+        itemView.favouriteFAB.setOnClickListener {
+            favouriteClickListener?.onClick(event.id, event.isFavourite)
+        }
+    }
+
+    fun setFabBackground(status: Boolean) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt]: status should be isFavorite

File: app/src/main/res/layout/item_card_events.xml [hunk 5]
```diff
@@ -20,13 +20,26 @@
...
             android:scaleType="center"
             app:elevation="@dimen/fab_elevation"
             app:backgroundTint="@color/white"
             app:srcCompat="@drawable/ic_share_grey_24dp"
             android:background="@color/white" />
 
+        <android.support.design.widget.FloatingActionButton
+            android:layout_width="@dimen/fab_width"
+            android:layout_height="@dimen/fab_height"
+            android:id="@+id/favouriteFAB"
```
Comment (iamareebjamal) [inline app/src/main/res/layout/item_card_events.xml]: Use camel case

**Comment by simarsingh24** [comment]
@iamareebjamal resolved

**Comment by nikit19** [comment]
Build is failing @simarsingh24

**Comment by simarsingh24** [comment]
@nikit19 because of "recycler animation removed" PR got merged.
Resolved now @iamareebjamal please merge this

File: app/src/main/java/org/fossasia/openevent/general/event/Event.kt [hunk 6]
```diff
@@ -61,5 +61,6 @@ data class Event(
         val isSessionsSpeakersEnabled: Boolean = false,
         val isTicketingEnabled: Boolean = false,
         val isTaxEnabled: Boolean = false,
-        val isMapShown: Boolean = false
+        val isMapShown: Boolean = false,
+        val favourite: Boolean = false
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/Event.kt]: We follow american spelling, `isFavorite`

**Comment by simarsingh24** [comment]
@iamareebjamal please merge this

**Comment by simarsingh24** [comment]
@iamareebjamal resolved

**Comment by simarsingh24** [comment]
@iamareebjamal please merge this so that i can work on #160

**Comment by simarsingh24** [comment]
@iamareebjamal please merge this

**Pull request merged**